### PR TITLE
Feature: onTransactionSent event listener

### PR DESCRIPTION
**Pull request by surewht**

### CODE REVIEW

File: packages/widget/src/utils.ts [hunk 1]
```diff
@@ -69,3 +69,18 @@ export function toFixedUsingString(numStr: string, decimalPlaces: number) {
 export type Prettify<T> = {
   [K in keyof T]: T[K];
 } & {};
+
+type ExtractABIFunctionNames<T, S> = T extends {
```
Comment (kasparkallas) [inline packages/widget/src/utils.ts:73]: Also available here: https://abitype.dev/api/utilities#extractabifunctionnames

File: packages/widget/src/ContractWriteManager.tsx [hunk 2]
```diff
@@ -49,6 +53,11 @@ export function ContractWriteManager({
       ? (prepareResult.config as unknown as ContractWrite)
       : contractWrite),
     onError: console.error,
+    onSuccess: ({ hash }, { functionName }) =>
+      eventListeners.onTransactionExecuted?.({
+        hash,
+        functionName: functionName as TxFunctionName,
```
Comment (kasparkallas) [inline packages/widget/src/ContractWriteManager.tsx:59]: Good for now. We'll be adding batch support, then will be best to pass "commands" or "commandNames" here as well.

File: apps/hosted-widget/pages/[hash].tsx [hunk 3]
```diff
@@ -93,7 +93,15 @@ const IPFSWidgetPage: NextPage = () => {
                     ajs.track("button_click", { type: props?.type }),
                   onRouteChange: (props) =>
                     ajs.track("route_change", { route: props?.route }),
-                  onSuccess: () => ajs.track("stream_started"),
+                  onTransactionExecuted: (props) => {
```
Comment (kasparkallas) [inline apps/hosted-widget/pages/[hash].tsx:96]: `onTransactionSent` is a bit more correct and aligned with termniology, IMO.

**Review by kasparkallas — COMMENTED** [review]
Couple of nitpicky comments, otherwise I like it. 👍🏼

File: packages/widget/src/utils.ts (hunk 1 above)
Comment (tokdaniel) [reply packages/widget/src/utils.ts]: added it, less pollution

**Review by kasparkallas — APPROVED** [review]

**Pull request merged**